### DISCUSSION TITLE
Lambda idempotency

### DISCUSSION THREAD

**Discussion by macdrorepo** [ran-isenberg/aws-lambda-handler-cookbook#583] [Ideas]
Hello,

Great examples in repository and blog entries, thank you for that!
If that would be possible could you add some idempotency for lambda functions? I know there are powertools docs but I do not feel the subject fully. Would be great to see that in some context like logging, tracing here.

**Comment by ran-isenberg** [comment]
great idea!
I'll add it to my todo list.
Thanks!

**Reply by ran-isenberg** [reply]
@macdrorepo  havent forgotten, been busy with conferences and work, it's still in my todo list!

**Reply by ran-isenberg** [reply]
@macdrorepo  see https://github.com/ran-isenberg/aws-lambda-handler-cookbook/pull/654
coming soon :)

**Reply by macdrorepo** [reply]
How soon ;) ?

**Comment by ran-isenberg** [comment]
It's already there.
Check also my blog post about it[
](https://www.ranthebuilder.cloud/post/serverless-api-idempotency-with-aws-lambda-powertools-and-cdk)
I explain the code I added to the cookbook
